code/modules/clothing/chameleon.dm

```diff
@@ -580,6 +580,39 @@
 	var/datum/action/item_action/chameleon/drone/randomise/randomise_action = new(src)
 	randomise_action.UpdateButtonIcon()
 
+/datum/action/item_action/chameleon/tongue_change
+	name = "Tongue Change"
+	icon_icon = 'icons/obj/surgery.dmi'
+	button_icon_state = "tonguebone"
+	var/list/tongue_list = list()
+
+/datum/action/item_action/chameleon/tongue_change/proc/generate_tongue_list()
+	for(var/V in typesof(/obj/item/organ/tongue))
+		if(ispath(V))
+			var/obj/item/I = V
+			if((initial(I.item_flags) & ABSTRACT) || !initial(I.icon_state))
+				continue
+			var/tongue_name = "[initial(I.name)] ([initial(I.icon_state)])"
+			tongue_list[tongue_name] = I
+
+/datum/action/item_action/chameleon/tongue_change/Trigger()
+	if(!IsAvailable())
+		return
+	if(!isitem(target))
+		return
+	var/obj/item/clothing/mask/M = target
+	var/obj/item/organ/tongue/picked_tongue
+	var/picked_name
+	picked_name = input("Select tongue to change into", "Chameleon tongue", picked_name) as null|anything in sort_list(tongue_list, GLOBAL_PROC_REF(cmp_typepaths_asc))
```

Comment:
From a player facing perspective, would it not make more sense to sort this by the tongues initial name?